components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/internal/builder/AuthReqBuilder.java

```diff
@@ -106,4 +185,150 @@ private static NameIDPolicy buildNameIDPolicy() {
+     * @param signatureAlgorithm
+     * @param cred
+     * @return
+     */
+    
+    
+    public static LogoutRequest setSignature(LogoutRequest logoutRequest, String signatureAlgorithm,
+            X509Credential cred) throws Exception {
+        try {
+            Signature signature = (Signature) buildXMLObject(Signature.DEFAULT_ELEMENT_NAME);
+            signature.setSigningCredential(cred);
+            signature.setSignatureAlgorithm(signatureAlgorithm);
+            signature.setCanonicalizationAlgorithm(SignatureConstants.ALGO_ID_C14N_EXCL_OMIT_COMMENTS);
+
+            try {
+                KeyInfo keyInfo = (KeyInfo) buildXMLObject(KeyInfo.DEFAULT_ELEMENT_NAME);
+                X509Data data = (X509Data) buildXMLObject(X509Data.DEFAULT_ELEMENT_NAME);
+                org.opensaml.xml.signature.X509Certificate cert =
+                        (org.opensaml.xml.signature.X509Certificate) buildXMLObject(
+                                org.opensaml.xml.signature.X509Certificate.DEFAULT_ELEMENT_NAME);
+                String value = Base64.encodeBytes(cred.getEntityCertificate().getEncoded());
+                cert.setValue(value);
+                data.getX509Certificates().add(cert);
+                keyInfo.getX509Datas().add(data);
+                signature.setKeyInfo(keyInfo);
+            } catch (CertificateEncodingException e) {
+                throw new SecurityException("Error getting certificate", e);
+            }
+
+            logoutRequest.setSignature(signature);
+
+            List<Signature> signatureList = new ArrayList<Signature>();
+            signatureList.add(signature);
+
+            // Marshall and Sign
+            MarshallerFactory marshallerFactory =
+                    org.opensaml.xml.Configuration.getMarshallerFactory();
+            Marshaller marshaller = marshallerFactory.getMarshaller(logoutRequest);
+
+            marshaller.marshall(logoutRequest);
+
+            Signer.signObjects(signatureList);
+            return logoutRequest;
+
+        } catch (Exception e) {
+            throw new Exception("Error while signing the Logout Request message", e);
+        }
+    }
+    
+
+    /**
+     * Builds SAML Elements
+     *
+     * @param objectQName
+     * @return
+     */
+    private static XMLObject buildXMLObject(QName objectQName) throws Exception {
+        XMLObjectBuilder builder =
+                org.opensaml.xml.Configuration.getBuilderFactory()
+                        .getBuilder(objectQName);
+        if (builder == null) {
+            throw new Exception("Unable to retrieve builder for object QName " +
+                    objectQName);
+        }
+        return builder.buildObject(objectQName.getNamespaceURI(), objectQName.getLocalPart(),
+                objectQName.getPrefix());
+    }
+
+    /**
+     * Replaces the ${} in url with system properties and returns
+     *
+     * @param acsUrl
+     * @return
+     */
+    private String processAcsUrl(String acsUrl){
```

Comment:
Fixed in #1581 (https://github.com/wso2/carbon-apimgt/commit/0e5a93cca133738cc74a26863424e20db485892d)